### ISSUE TITLE
Add Rule Requiring ref in dbt Models

### ISSUE THREAD

**Issue by erika-e** [sqlfluff/sqlfluff#2869]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

In dbt models writing explicit table references like the below is an antipattern. Doing so means dbt cannot properly resolve DAG dependencies.

```SQL
select
*
from database.schema.table
```
The preferred way to write the above in a dbt model would be:
```SQL
select
*
from {{ ref('table') }}
```

It would be fantastic to have a SQLfluff rule to lint for this.

### Use case

I often run lint or fix after modifying a model during my development process. Lately I've been forgetting to replace explicit references to tables with dbt ref functions. I'd rather this be flagged by the linter than in PR review.

In my use case, I'd be ok with `sqlfluff fix` replacing the explicit reference with the equivalent ref, though I can see situations where this might not be the desired behaior.

### Dialect

We use the dbt templater and snowflake dialect, although I can see this being broadly applicable.

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by barrywhart** [comment]
Hi @erika-e, you mentioned you would look into adding this rule. Have you started yet? How are things going?
